code/search/SearchCriteria.php

```diff
@@ -0,0 +1,233 @@
+            $clause->setAdapter($adapter);
+            $clause->appendPreparedStatementTo($ps);
+
+            // There's always one less conjunction then there are clauses.
+            if ($this->getConjunction($key) !== null) {
+                $ps .= $adapter->getConjunctionFor($this->getConjunction($key));
+            }
+        }
+
+        $ps .= $adapter->getCloseComparisonContainer();
+    }
+
+    /**
+     * @param string|SearchCriteriaInterface $target
+     * @param mixed $value
+     * @param string|null $comparison
+     * @param AbstractSearchQueryWriter $searchQueryWriter
+     * @return $this
+     */
+    public function addAnd(
+        $target,
+        $value = null,
+        $comparison = null,
+        AbstractSearchQueryWriter $searchQueryWriter = null
+    ) {
+        $criterion = $this->getCriterionForCondition($target, $value, $comparison, $searchQueryWriter);
+
+        $this->addConjunction(SearchCriteria::UND);
+        $this->addClause($criterion);
+
+        return $this;
+    }
+
+    /**
+     * @param string|SearchCriteriaInterface $target
+     * @param mixed $value
+     * @param string|null $comparison
+     * @param AbstractSearchQueryWriter $searchQueryWriter
+     * @return $this
+     */
+    public function addOr(
+        $target,
+        $value = null,
+        $comparison = null,
+        AbstractSearchQueryWriter $searchQueryWriter = null
+    ) {
+        $criterion = $this->getCriterionForCondition($target, $value, $comparison, $searchQueryWriter);
+
+        $this->addConjunction(SearchCriteria::ODER);
+        $this->addClause($criterion);
+
+        return $this;
+    }
+
+    /**
+     * @param string|SearchCriteriaInterface $target
+     * @param mixed $value
+     * @param string $comparison
+     * @param AbstractSearchQueryWriter $searchQueryWriter
+     * @return SearchCriteriaInterface
+     */
+    protected function getCriterionForCondition(
+        $target,
+        $value,
+        $comparison,
+        AbstractSearchQueryWriter $searchQueryWriter = null
+    ) {
+        if ($target instanceof SearchCriteriaInterface) {
+            return $target;
+        }
+
+        return new SearchCriterion($target, $value, $comparison, $searchQueryWriter);
+    }
+
+    /**
+     * @return SearchCriteriaInterface[]
+     */
+    protected function getClauses()
+    {
+        return $this->clauses;
+    }
+
+    /**
+     * @param SearchCriteriaInterface $criterion
+     */
+    protected function addClause($criterion)
+    {
+        $this->clauses[] = $criterion;
+    }
+
+    /**
+     * @return string[]
+     */
+    protected function getConjunctions()
+    {
+        return $this->conjunctions;
+    }
+
+    /**
+     * @param int $key
+     * @return string|null
+     */
+    protected function getConjunction($key)
+    {
+        if (!array_key_exists($key, $this->getConjunctions())) {
+            return null;
+        }
+
+        return $this->getConjunctions()[$key];
```

Comment:
This module still supports PHP 5.3, which doesn't support array dereferencing. Could you please assign `$this->getConjunctions()` to a variable and use that?